src/main/java/algorithms/curated170/hard/EncodeStringWithShortestLengthDP.java

```diff
@@ -0,0 +1,53 @@
+package algorithms.curated170.hard;
+
+public class EncodeStringWithShortestLengthDP {
```

Comment:
We may find some ways to simplify this code, I think. It is hard to fallow.

Reply:
Is it more clear now?